pandas/io/pytables.py

```diff
@@ -2960,11 +2960,17 @@ def write_array_empty(self, key: str, value: ArrayLike):
         node._v_attrs.value_type = str(value.dtype)
         node._v_attrs.shape = value.shape
 
-    def write_array(self, key: str, value: ArrayLike, items: Optional[Index] = None):
-        # TODO: we only have one test that gets here, the only EA
+    def write_array(self, key: str, obj, items: Optional[Index] = None):
```

Comment:
why is the typing changed? can you be explicit here

Reply:
Added a type hint.

AFAIU for series `to_numpy()` drops timezone info so we need to pass the original object in `SeriesFixed.read(...)`and special case here on timezone-aware data

xref (from stale PR that worked on this) https://github.com/pandas-dev/pandas/pull/20595/files#r179816447